tests/test_http_parser.py

```diff
@@ -787,6 +788,70 @@ def test_url_parse_non_strict_mode(parser) -> None:
     assert payload.is_eof()
 
 
+@pytest.mark.parametrize(
+    'uri,path,query,fragment',
```

Comment:
I've recently discovered that you can make the params a bit more readable in pytest:
```suggestion
    ('uri', 'path', 'query', 'fragment'),
```